src/components/Savings/SavingsCard/SavingsCardTableMobile/index.tsx

```diff
@@ -9,95 +9,119 @@ import { ModalType } from 'components/Savings/SavingsModal'
-                                                    case 'principle':
-                                                        return (
-                                                            <div>
-                                                                {stats?.[item.statsKey]?.format({
-                                                                    useFixedPrecision: true,
-                                                                    fixedPrecisionDigits: 2,
-                                                                })}
-                                                            </div>
-                                                        )
-                                                    case 'claimable':
-                                                        return (
-                                                            <>
-                                                                <div>{stats?.[item.statsKey]?.g$Reward.format()}</div>
-                                                                <div>{stats?.[item.statsKey]?.goodReward.format()}</div>
-                                                            </>
-                                                        )
-                                                    default:
-                                                        return
-                                                }
-                                            })()
-                                        )}
+        <>
+            {hasBalance && (
+                <Card className="mb-6 md:mb-4 card">
+                    <CellSC className="h-72">
+                        <div className="flex flex-row flex-wrap items-center justify-start max-w-xs gap-x-16 gap-y-4">
+                            {headings.map(
+                                (item, index) =>
+                                    index !== 0 && ( // skip type header on mobile, in line with stake positions
+                                        <div key={index}>
+                                            <div className="flex flex-grow-1">
+                                                <Title type="category" className="flex items-center">
+                                                    {item.title} <QuestionHelper text={item.questionText || ''} />
+                                                </Title>
+                                            </div>
+                                            <div className="font-bold value">
+                                                {error ? (
+                                                    <LoadingPlaceHolder />
+                                                ) : (
+                                                    stats &&
+                                                    hasBalance &&
+                                                    (() => {
```

Comment:
I already reviewed something like this one and suggested to create nested component.
as I remember this was PR about merging some stat tables for fuse and cell into a single one, Your and @roma-hladilka were working on it. please check in history.

- if this was already done, you've lost those changes during master merge  onto Your PR
- if those components could be reused they should do
- otherwise let's discuss tomorrow at sync